logpuzzle.py

```diff
@@ -21,15 +23,48 @@
 import sys
 import urllib
 import argparse
+import shutil
+
+html_template = """
+<html>
+<head>
+</head>
+<body>
+{}
+</body>
+</html>
+"""
+
+
+def url_sort_func(url_list):
+    """Sorts URLs based on last section of letters"""
+    sort_list = re.search(r'(/p-\w+-)(\w*)', url_list)
+    return sort_list.group(2)
+
+
+def create_img_template(path):
+    """Create's image template to put into html"""
+    return '<img src="{}"/>'.format(path)
 
 
 def read_urls(filename):
     """Returns a list of the puzzle urls from the given log file,
     extracting the hostname from the filename itself.
     Screens out duplicate urls and returns the urls sorted into
     increasing order."""
-    # +++your code here+++
-    pass
+
+    with open(filename, 'r') as f:
+
+        file_text = f.read()
+        match_list = re.findall(r'GET (\S*puzzle\S*)', file_text)
+
+        if re.search(r'(/p-\w+-)(\w*)', file_text):
+            new_match_list = sorted(['https://code.google.com'
+                                    + url for url in set(match_list)])
```

Comment:
if you sorted the set before concatenating the base url, it would save that many characters from comparison during sorting